### PR TITLE
Refactor Welcome Email & Settings file

### PR DESCRIPTION
**Pull request by toinevk**
Addresses #134 & #132. 132 was straight forward adding both a check to oath & email url. 

Preventing welcome email was slightly more tricky. Given OAuth emails are verified by default, its still appropriate to send out during on create user call. 

For non OAuth users, email should be sent once verification is successful (i.e. Client side). In addition, during verification call we do not have access to the user object and it must be called server side (using Meteor.user).

### CODE REVIEW

File: imports/api/Users/server/methods.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Accounts } from 'meteor/accounts-base';
 import editProfile from './edit-profile';
 import exportData from './export-data';
 import deleteAccount from './delete-account';
+import sendWelcomeEmail from './/send-welcome-email';
```
Comment (tomscholz) [inline imports/api/Users/server/methods.js]: There is a typo here. It should be `import sendWelcomeEmail from './send-welcome-email';`
Just remove one of the two slashes

File: imports/startup/server/accounts/oauth.js [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
+  if (OAuthSettings) {
+    Object.keys(OAuthSettings).forEach((service) => {
+      ServiceConfiguration.configurations.upsert(
+        { service },
+        { $set: OAuthSettings[service] },
+      );
+    });
+  }
+} else {
+  console.warn('Warning! Oauth settings are not configured. OAuth logins will not function.');
```
Comment (tomscholz) [inline imports/startup/server/accounts/oauth.js]: Typo: Oauth --> OAuth

File: imports/startup/server/email.js [hunk 3]
```diff
@@ -1,3 +1,7 @@
 import { Meteor } from 'meteor/meteor';
 
-if (Meteor.isDevelopment) process.env.MAIL_URL = Meteor.settings.private.MAIL_URL;
+if (Meteor.settings.private && Meteor.settings.private.MAIL_URL) {
+  if (Meteor.isDevelopment) process.env.MAIL_URL = Meteor.settings.private.MAIL_URL;
+} else {
+  console.warn('Warning! You have not configured mail settings. Emails will not be sent.');
```
Comment (tomscholz) [inline imports/startup/server/email.js]: Suggestion: Change `mail` and `Email` to `e-mail`. That way it's more consistent

**Comment by toinevk** [comment]
Thx!

**Comment by cleverbeagle** [comment]
Merging manually. Ran into a few undefined errors inside of the `send-welcome-email.js` module but have corrected these.

**Pull request closed without merging**